addons/account_peppol/models/account_edi_proxy_user.py

```diff
@@ -29,6 +31,34 @@ def _get_proxy_urls(self):
         }
         return urls
 
+    @api.model
+    def _get_peppol_error_message(self, error_vals):
+        """
+        Helper to process the error dictionary returned from the IAP response.
+        It will only get the code (or EBMS code) and map it to the correct translated message.
+        :param dict error_vals: the dictionary of encoded error json generated from the `_json` method in `peppol_proxy`
+        :return: the translated error message
+        :rtype: str
+        """
+        default_error_message = _('Unknown Peppol Error: %s', error_vals)
+
+        if (ebms_code := error_vals.get('ebms_code')) and ebms_code != 4:
+            # Error with ebMS code is originally from PeppolInboundError
+            # In most case, ebMS message will be better and more specific, except for when the code is 4 (general "Other" message)
+            error_message = get_ebms_message(ebms_code, default_error_message)
+        else:
+            error_message = get_exception_message(error_vals['code'], default_error_message)
+
+        if (msg_args_count := error_message.count("%s")) and msg_args_count == len(error_vals['args']):
+            error_message = error_message % tuple(error_vals['args'])
```

Comment:
This prevents the fallback mechanism to work :thinking:  (see [Coding guidelines](https://www.odoo.com/documentation/master/contributing/development/coding_guidelines.html#use-translation-method-correctly))
@dylankiss Can you help us find a better solution here ? :pray:

Reply:
What if you were to make your exception message getter functions work like this?
```python
def get_exception_message(peppol_code, default_message="", *args):
    match peppol_code:
        case 101:
            return _("Something went wrong with your request")
        case 102:
            return _('Proxy error, please contact Odoo (missing "%s" - please make sure that it was loaded from the configuration panel)', *args)
        ...
        case _:
            return default_message
```
and then call them like
```python
get_exception_message(error_vals['code'], default_message=default_error_message, *error_vals['args'])
```

Reply:
We can even probably keep the map if YONI prefers, and that will be a good usage of the lambda :p